File: android/src/main/java/com/getcapacitor/community/classes/events/FailToEstablishSecureConnectionEvent.java
package com.getcapacitor.community.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import com.getcapacitor.community.BridgefyController.Reason;
import java.util.UUID;

public class FailToEstablishSecureConnectionEvent extends FailToEvent {

    @NonNull
    UUID userID;

    public FailToEstablishSecureConnectionEvent(@NonNull UUID userID, @NonNull Reason reason) {
        super(reason);
        this.userID = userID;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        result.put("userID", userID.toString());
        result.put("reason", super.toJSObject());

        return result;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Events/FailToEstablishSecureConnectionEvent.swift
import Foundation
import Capacitor

@objc public class FailToEstablishSecureConnectionEvent: FailToEvent {
    let userID: UUID

    init(_ userID: UUID, _ error: BridgefyError?) {
        self.userID = userID

        super.init(error)
    }

    override public func toJSObject() -> JSObject {
        var result = JSObject()

        result["userID"] = userID.uuidString.lowercased()
        result["reason"] = super.toJSObject()

        return result
    }
}
